src/components/LoadDatasetModal.js

```diff
@@ -53,8 +55,16 @@ class LoadDatasetModal extends Component {
     dispatchAddDataset: PropTypes.func.isRequired,
   };
 
+  buildDefaultDatasetName = (newFileLocation) => {
+    return path.basename(newFileLocation);
+  };
```

Comment:
We should also remove the `.json` extension from the name.
Also remove the default name logic from the listener `loadDataset.js` since it is done here.